cmd/snap/cmd_snap_op.go

```diff
@@ -284,7 +284,7 @@ func showDone(names []string, op string) error {
 		default:
 			fmt.Fprintf(Stdout, "internal error: unknown op %q", op)
 		}
-		if snap.TrackingChannel != snap.Channel {
+		if snap.TrackingChannel != snap.Channel && snap.Channel != "" {
```

Comment:
I understand this change, but I don't understand its presence on this PR. Drive-by? Bad merge? (logic seems fine though)

Reply:
the new api will return an empty channel when doing install by revision, the old API was just returning a bit of a non-sense guess. Some part of this code were already checking for snap:Channel empty but not this one.